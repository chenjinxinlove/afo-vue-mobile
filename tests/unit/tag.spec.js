import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Tag from '../../src/tag'

describe('Tag.vue', () => {
  // 测试是否可以被使用
  it('use', () => {
    Vue.use(Tag)
    expect(Vue.component(Tag.name))
      .toHaveProperty('name', 'VueComponent')
  })
  // 测试props
  it('props', () => {
    const wrapper = shallowMount(Tag, {
      propsData: {
        type: 'primary',
        plain: true
      }
    })
    expect(wrapper.classes()).toEqual(['afo-tag--primaryplain', 'afo-tag'])
  })
  // 测试props
  it('props', () => {
    const wrapper = shallowMount(Tag, {
      propsData: {
        mark: true
      }
    })
    expect(wrapper.classes()).toEqual(['afo-tag--default', 'afo-tag--mark', 'afo-tag'])
  })
})
