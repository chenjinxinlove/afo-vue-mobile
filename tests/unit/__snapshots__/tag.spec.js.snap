// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag.vue demo snapshot 1`] = `
<div>
  <div>
    <div class="afo-tag--default afo-tag">标签</div>
    <div class="afo-tag--danger afo-tag">标签</div>
    <div class="afo-tag--primary afo-tag">标签</div>
  </div>
  <div>
    <div class="afo-tag--defaultplain afo-tag">标签</div>
    <div class="afo-tag--dangerplain afo-tag">标签</div>
    <div class="afo-tag--primaryplain afo-tag">标签</div>
  </div>
  <div>
    <div class="afo-tag--default afo-tag--mark afo-tag">标签</div>
    <div class="afo-tag--danger afo-tag--mark afo-tag">标签</div>
    <div class="afo-tag--primary afo-tag--mark afo-tag">标签</div>
  </div>
</div>
`;
